python/ray/data/_internal/logical/operators/all_to_all_operator.py

```diff
@@ -53,12 +53,13 @@ class RandomShuffle(AbstractAllToAll):
     def __init__(
         self,
         input_op: LogicalOperator,
+        name: str = "RandomShuffle",
         seed: Optional[int] = None,
         num_outputs: Optional[int] = None,
         ray_remote_args: Optional[Dict[str, Any]] = None,
     ):
         super().__init__(
-            "RandomShuffle",
+            name,
```

Comment:
this is needed so we can override the name with the upstream operator-fused name.